### PR TITLE
Add the first endpoint: tags

### PR DESCRIPTION
**Pull request by zupo**
This also means that db is now properly integrated and tested, along
with db schema migrations with alembic.

I've also added a bunch of tests and type hints to hit 100% for both
tests and type hints coverages.

### CODE REVIEW

File: src/conduit/__init__.py [hunk 1]
```diff
@@ -31,16 +37,45 @@ def configure_logging() -> None:
...
+        sys.exit(f"ERROR: DB ({curr}) is not migrated to head ({head}). Shutting down.")
+
+
 def configure(config: Configurator) -> None:
     """Configure Pyramid to serve the Conduit API."""
 
     # Configure DB
     config.include("pyramid_deferred_sqla")
     config.sqlalchemy_engine(pool_size=5, max_overflow=1, pool_timeout=5)
+    config.alembic_config("conduit]:migrations")
```
Comment (am-on) [inline src/conduit/__init__.py]: typo?

File: src/conduit/tests/test_check_db_migrated.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
+"""Test def check_db_migrated function."""
+
+from conduit import check_db_migrated
+from conduit import main
+from unittest import mock
+
+
+@mock.patch("conduit.alembic")
+def test_skip_when_running_an_alembic_command(alembic):
```
Comment (sayanarijit) [inline src/conduit/tests/test_check_db_migrated.py]: Is this file intentionally not hinted?

File: src/conduit/tag/views.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+from conduit.tag.models import Tag
+from pyramid.request import Request
+from pyramid.view import view_config
+
+import typing as t
+
+
+@view_config(route_name="tags", renderer="json", request_method="GET", openapi=True)
+def tags(request: Request) -> t.Dict:
```
Comment (sayanarijit) [inline src/conduit/tag/views.py]: Maybe ```t.Dict[str, t.List[Tag]]```

File: src/conduit/tests/test_check_db_migrated.py (hunk 2 above)
Comment (zupo) [reply src/conduit/tests/test_check_db_migrated.py]: Hmmm. It is not. I wonder why mypy's coverage didn't report it.

Good catch!

File: src/conduit/tests/test_check_db_migrated.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    global_config = None
+
+    assert check_db_migrated(config, global_config) is None
+
+
+@mock.patch("conduit.alembic")
+@mock.patch("conduit.EnvironmentContext")
+@mock.patch("conduit.ScriptDirectory")
+@mock.patch("conduit.MigrationContext")
+@mock.patch("conduit.sys")
```
Comment (dz0ny) [inline src/conduit/tests/test_check_db_migrated.py]: Use mocker fixture

Comment (zupo) [reply src/conduit/tests/test_check_db_migrated.py]: I did. I really don't see the point. Is there a blog post that shows the benefits?

File: src/conduit/tests/test_check_db_migrated.py (hunk 2 above)
Comment (zupo) [reply src/conduit/tests/test_check_db_migrated.py]: It's because `tests` folders don't have `__init__.py` files. Makes sense. Adding them now.

**Pull request merged**